inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/model/KnowledgeBase.java

```diff
@@ -416,4 +420,17 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(repositoryId, name);
     }
+    
+    public void applyRootConcept(KnowledgeBaseMapping aMapping) {
+        if (!aMapping.getRootConcepts().isEmpty()
+                && !(aMapping.getRootConcepts().contains(OWL.NOTHING.stringValue()))) {
+            if (explicitlyDefinedRootConcepts.isEmpty()) {
```

Comment:
I do not understand why the conditions in lines 425-427 are required. IMHO it would be sufficient to simply have the code from line 428-431 plus a line before that which cleans out any existing explicitlyDefinedRootConcepts. Line 432 seems redundant. I also do not understand why `applyRootConcept` is a separate method - I believe it shouldn't be called independently of `applyMapping`.

Reply:
Because we don't want to overwrite the value of the root-concepts if the user explicitly provides root-concept in the configuration settings. What we can do is, remove `OWL.NOTHING` from the mapping file in yaml and set that to pass an empty list. We can remove the rest of the condition barring `if (explicitlyDefinedRootConcepts.isEmpty())` .

Reply:
> Because we don't want to overwrite the value of the root-concepts if the user explicitly provides root-concept in the configuration settings.

Hmm... If the root concepts are part of the pre-configured mapping, then I think it would be strange not to add these root concepts when the mapping is selected. But I also understand that it could be annoying to the user to manually configure some root concepts and loosing them all when selecting some other pre-configured mapping. 

At least this behavior should be documented in the user documentation and there should be a comment in the code as well explaining the rationale behind this.